packages/syft/src/syft/core/tensor/fixed_precision_tensor.py

```diff
@@ -4,83 +4,126 @@
 # stdlib
 from typing import Any
 from typing import Optional
+from typing import Union
 
 # third party
-from google.protobuf.reflection import GeneratedProtocolMessageType
 import numpy as np
 
 # relative
-from ...proto.core.tensor.fixed_precision_tensor_pb2 import (
-    FixedPrecisionTensor as FixedPrecisionTensor_PB,
-)  # type: ignore
-from ..common.serde.deserialize import _deserialize as deserialize
 from ..common.serde.serializable import serializable
-from ..common.serde.serialize import _serialize as serialize
 from .passthrough import PassthroughTensor  # type: ignore
+from .passthrough import is_acceptable_simple_type  # type: ignore
+from .smpc import context
 
 
-@serializable()
+@serializable(recursive_serde=True)
 class FixedPrecisionTensor(PassthroughTensor):
+
+    __attr_allowlist__ = ("child", "_base", "_precision", "_scale")
+
     def __init__(
-        self, value: Optional[Any] = None, base: int = 10, precision: int = 3
+        self, value: Optional[Any] = None, base: int = 2, precision: int = 9
     ) -> None:
+
         self._base = base
         self._precision = precision
         self._scale = base**precision
         if value is not None:
+            # TODO :Should modify to be compatiable with torch.
+            value = np.array(value, dtype=np.int32)
             fpt_value = self._scale * value
-            encoded_value = fpt_value.astype(np.int64)
+            encoded_value = fpt_value.astype(np.int32)
```

Comment:
Q + MAYBE: Could we extract this np.int32 and maybe also the ring size to a variable in `sympc/defaults.py`? (something like `DEFAULT_FPT_NUMPY_TYPE` and `DEFAULT_RING_SIZE`?)

I am thinking that maybe at one point we might change one and omit the others

Reply:
Modified